packages/arb-util/broadcaster/clientconnection.go

```diff
@@ -36,34 +36,36 @@ const MaxSendQueue = 20
 
 // ClientConnection represents client connection.
 type ClientConnection struct {
-	ioMutex sync.Mutex
-	conn    io.ReadWriteCloser
+	parentCtx context.Context
+	ioMutex   sync.Mutex
+	conn      io.ReadWriteCloser
 
 	desc          *netpoll.Desc
 	name          string
 	clientManager *ClientManager
 
 	timeoutMutex sync.Mutex
 	lastHeard    time.Time
-	cancelFunc context.CancelFunc
-	out        chan []byte
+	cancelFunc   context.CancelFunc
+	out          chan []byte
 }
 
-func NewClientConnection(conn net.Conn, desc *netpoll.Desc, clientManager *ClientManager) *ClientConnection {
+func NewClientConnection(ctx context.Context, conn net.Conn, desc *netpoll.Desc, clientManager *ClientManager) *ClientConnection {
 	return &ClientConnection{
+		parentCtx:     ctx,
 		conn:          conn,
 		desc:          desc,
 		name:          conn.RemoteAddr().String() + strconv.Itoa(rand.Intn(10)),
 		clientManager: clientManager,
 		lastHeard:     time.Now(),
+		out:           make(chan []byte, MaxSendQueue),
 	}
 }
 
-func (cc *ClientConnection) Start(parentCtx context.Context) {
-	ctx, cancelFunc := context.WithCancel(parentCtx)
+func (cc *ClientConnection) Start() {
+	ctx, cancelFunc := context.WithCancel(cc.parentCtx)
```

Comment:
Since `Start()` is called in the ClientManager thread, it would get a different ctx than it was originally passed in.  In this case, it doesn't matter, so I will update it to make it simpler

Reply:
I think having the context passed in as tightly scoped to the operation that's being controlled by the context tends to work best. If the constructor needs a context that is only used for operations that are synchronous in the constructor, no need to use the same context here

Reply:
I agree, was overthinking things.